src/xcode/ENA/ENA/Source/Scenes/ContactDiary/Store/ContactDiaryStore.swift

```diff
@@ -505,15 +504,109 @@ class ContactDiaryStore: DiaryStoring, DiaryProviding {
 	}
 
 	@discardableResult
-	func updateContactPersonEncounter(id: Int, contactPersonId: Int, date: String, duration: ContactPersonEncounter.Duration?, maskSituation: ContactPersonEncounter.MaskSituation?, setting: ContactPersonEncounter.Setting?, circumstances: String) -> DiaryStoringVoidResult {
-		// CJE: Implement
-		return .success(())
+	func updateContactPersonEncounter(
+		id: Int,
+		date: String,
+		duration: ContactPersonEncounter.Duration,
+		maskSituation: ContactPersonEncounter.MaskSituation,
+		setting: ContactPersonEncounter.Setting,
+		circumstances: String
+	) -> DiaryStoringVoidResult {
+		var result: DiaryStoringVoidResult?
+
+		databaseQueue.inDatabase { database in
+			Log.info("[ContactDiaryStore] Update ContactPersonEncounter with id: \(id).", log: .localData)
+
+			let sql = """
+				UPDATE ContactPersonEncounter
+				SET date = SUBSTR(?, 1, 250), duration = ?, maskSituation = ?, setting = ?, circumstances = SUBSTR(?, 1, 250)
```

Comment:
I personally would have handled that in code not SQL. E.g. changing this limit requires to look up for all occurrences of the '250' value. Just mentioning.

Reply:
Good point. I have added a constant for the max text length.